lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/AbstractPostgresKgStore.scala

```diff
@@ -0,0 +1,91 @@
+package io.github.tetherlessworld.mcsapps.lib.kg.stores.postgres
+
+import io.github.tetherlessworld.mcsapps.lib.kg.models.edge.KgEdge
+import io.github.tetherlessworld.mcsapps.lib.kg.models.node.KgNode
+import io.github.tetherlessworld.mcsapps.lib.kg.models.source.KgSource
+import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
+import slick.jdbc.PostgresProfile
+
+
+abstract class AbstractPostgresKgStore(protected val dbConfigProvider: DatabaseConfigProvider) extends HasDatabaseConfigProvider[PostgresProfile] {
+  import profile.api._
+
+  lazy val KgNodes = TableQuery[KgNodeTable]
+  lazy val KgEdges = TableQuery[KgEdgeTable]
+  lazy val KgSources = TableQuery[KgSourceTable]
+
+  private class KgNodeTable(tag: Tag) extends Table[KgNode](tag, "KgNodes") {
+    def id = column[String]("ID", O.PrimaryKey)
```

Comment:
I came up in the school that you should always have "id" be an autoincrement integer primary key and have separate UNIQUE constraints on "natural" keys like what we have in node.id. But others argue that it's fine to just have the "natural" key be the primary key if you have one. So you can leave this.

Reply:
Thinking about it, that makes sense. I'm going to go ahead and change to integer primary keys to see how that pans out